tuf/ngclient/updater.py

```diff
@@ -229,6 +231,9 @@ def download_target(
             Local path to downloaded file
         """
 
+        if self.config.offline:
+            raise exceptions.DownloadError("Cannot download when offline")
```

Comment:
I didn't introduce a new exception subclass since the context seems to make this useful enough for apps: if your updater is offline, then DownloadError means _the requested action is not possible with the currently cached data_.